pkg/pbq/store/wal/bootstrap.go

```diff
@@ -0,0 +1,223 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package wal
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/gob"
+	"fmt"
+	"go.uber.org/zap"
+	"io"
+	"os"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/isb"
+	"github.com/numaproj/numaflow/pkg/pbq/partition"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+)
+
+var location *time.Location
+
+func init() {
+	var err error
+	location, err = time.LoadLocation("UTC")
+	if err != nil {
+		panic(fmt.Sprint("cannot load UTC", err))
+	}
+}
+
+// IsCorrupted checks whether the file is corrupt
+func (w *WAL) IsCorrupted() bool {
+	return w.corrupted
+}
+
+// OpenWAL returns a WAL if present
+func OpenWAL(ctx context.Context, filePath string) (*WAL, error) {
+	stat, err := os.Stat(filePath)
+	// there could be other errors
+	if os.IsNotExist(err) {
+		return nil, nil
+	} else if err != nil {
+		log := logging.FromContext(ctx).With("pbqStore", "FileSystem")
+		log.Errorw("Error when opening WAL file", zap.Any("file name", filePath))
```

Comment:
BTW - use single word `fileName` as the log key - it will be structured as JSON.